cmd/contact-auditor/main.go

```diff
@@ -101,33 +110,38 @@ func (c contactAuditor) run(resChan chan *result) error {
 	}
 
 	for rows.Next() {
-		var res result
-		err := rows.Scan(&res.ID, &res.Contact)
+		var id int64
+		var contact []byte
+		var createdAt string
+		err := rows.Scan(&id, &contact, &createdAt)
 		if err != nil {
 			return err
 		}
 
-		err = res.unmarshalContact()
+		contacts, err := unmarshalContact(contact)
 		if err != nil {
-			c.logger.Errf("Unmarshal failed for ID: %d due to: %s", res.ID, err)
+			if c.writeToStdout || c.resultsFile != nil {
+				c.writeResults(fmt.Sprintf("%d\t%s\tunmarshal\t%q\t%q\n", id, createdAt, contact, err))
+
+			}
```

Comment:
This duplicates the logic inside `writeResults`. I would just call `writeResults` unconditionally, and let it decide what to do.

Reply:
🤦🏻‍♀️